Sources/AardvarkTests/ViewHierarchyAttachmentGeneratorTests.swift

```diff
@@ -0,0 +1,170 @@
+
+final class ViewHierarchyAttachmentGeneratorTests: XCTestCase {
+
+    func testAttachmentMetadata() {
+        let attachment = ViewHierarchyAttachmentGenerator.generateAttachment(for: [])
+
+        XCTAssertEqual(attachment.fileName, "view_hierarchy.txt")
+        XCTAssertEqual(attachment.dataMIMEType, "text/plain")
+    }
+
+    func testDescriptionForSimpleWindow() {
+        let window = TestWindow()
+        let attachment = ViewHierarchyAttachmentGenerator.generateAttachment(for: [window])
+        let description = String(data: attachment.data, encoding: .utf8)
+
+        XCTAssertEqual(
+            description,
+            """
+            Window Description
+
+            """
+        )
+    }
+
+    func testDescriptionForWindowWithSubviews() {
+        let window = TestWindow()
+
+        let firstSubview = TestView(identifier: "B")
+        window.addSubview(firstSubview)
+        firstSubview.addSubview(TestView(identifier: "C"))
+
+        window.addSubview(TestView(identifier: "A"))
+
+        let attachment = ViewHierarchyAttachmentGenerator.generateAttachment(for: [window])
+        let description = String(data: attachment.data, encoding: .utf8)
+
+        XCTAssertEqual(
+            description,
+            """
+            Window Description
+               | View Description for B
+               |    | View Description for C
+               | View Description for A
+
+            """
+        )
+    }
+
+    func testDescriptionForWindowWithSublayers() {
+        let window = TestWindow()
+
+        let subview = TestView(identifier: "Subview")
+        window.addSubview(subview)
+
+        subview.layer.addSublayer(TestLayer())
+        subview.layer.addSublayer(TestLayer())
+
+        let attachment = ViewHierarchyAttachmentGenerator.generateAttachment(for: [window])
+        let description = String(data: attachment.data, encoding: .utf8)
+
+        XCTAssertEqual(
+            description,
+            """
+            Window Description
+               | View Description for Subview
+               |    | Layer Description
+               |    | Layer Description
+
+            """
+        )
+    }
+
+    func testDescriptionForWindowWithViewController() {
+        let window = TestWindow()
+
+        let viewController = TestViewController()
+        window.rootViewController = viewController
+        window.addSubview(viewController.view)
+
+        let attachment = ViewHierarchyAttachmentGenerator.generateAttachment(for: [window])
+        let description = String(data: attachment.data, encoding: .utf8)
+
+        XCTAssertEqual(
+            description,
+            """
+            Window Description
+               | VC: View Controller Description
+               | View Description for Managed View
+
+            """
+        )
+    }
+
+}
+
+// MARK: -
+
+private final class TestWindow: UIWindow {
+
+    override var description: String {
+        return "Window Description"
```

Comment:
nit, no need for the `return`!
```suggestion
        "Window Description"
```